app/models/printers_user.rb

```diff
@@ -0,0 +1,4 @@
+class PrintersUser < ActiveRecord::Base
+  belongs_to :printer
+  belongs_to :user
+end
```

Comment:
Add a unique constraint validation so that you can't have the same user and printer listed twice for the same tag

`validates :role_id, uniqueness: { scope: [:user_id, :tag }`

See https://apidock.com/rails/ActiveRecord/Validations/ClassMethods/validates_uniqueness_of

Reply:
@abulsayyad123 did you add the constraint?

Reply:
Can you also validate that both printer_id and user_id exist? We don't want `nil` appearing anywhere